### PR TITLE
Converted AreaData & MonsterData to dynamic objects; Moved experience…

### PR DESCRIPTION
**Pull request by Nishimura-Katsuo**
… related tables to Misc object; Improved calculations of monsterExp and areaPartyExp

### CODE REVIEW

File: d2bs/kolbot/libs/GameData.js [hunk 1]
```diff
@@ -4,57 +4,173 @@
 *	@desc      game data library
 */
 
-/* globals AreaData, MonsterData, ExpCurve, ExpPenalty, MonsterExp */
+const MONSTERINDEXCOUNT = 734;
```
Comment (noah-) [inline d2bs/kolbot/libs/GameData.js]: separate words with _
like MONSTER_INDEX_COUNT

File: d2bs/kolbot/libs/GameData.js [hunk 2]
```diff
@@ -4,57 +4,173 @@
...
+ *  .Fire = fire resist
+ *  .Lightning = lightning resist
+ *  .Poison = poison resist
+ *  .Minions = array of minions that can spawn with this unit
+ */
+
+var MonsterData = Array(MONSTERINDEXCOUNT);
+
+for (let i = 0; i < MonsterData.length; i++) {
+	let index = i;
```
Comment (noah-) [inline d2bs/kolbot/libs/GameData.js]: why do we need a copy of i? why not just use i?

File: d2bs/kolbot/libs/common/Misc.js [hunk 3]
```diff
@@ -28,7 +28,6 @@ var Skill = {
 		case 112: // Blessed Hammer
 		case 116: // Conversion
 		case 126: // Bash
-		case 130: // Howl
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Misc.js]: why is this deleted?

File: d2bs/kolbot/libs/GameData.js (hunk 2 above)
Comment (Nishimura-Katsuo) [reply d2bs/kolbot/libs/GameData.js]: it's a scoping nuance, index is scoped at each iteration, and i isn't
if I don't do this then each new object I create refers to the last value i had instead of the value of index set for this iteration

File: d2bs/kolbot/libs/common/Misc.js (hunk 3 above)
Comment (Nishimura-Katsuo) [reply d2bs/kolbot/libs/common/Misc.js]: it's a duplicate,130/howl case is defined further down with specific calculations

File: d2bs/kolbot/libs/common/Misc.js [hunk 4]
```diff
@@ -69,7 +68,6 @@ var Skill = {
 		case 101: // Holy Bolt
 		case 107: // Charge
 		case 130: // Howl
-		case 132: // Leap
```
Comment (Nishimura-Katsuo) [inline d2bs/kolbot/libs/common/Misc.js]: same here, 132/leap was specified twice, but has more accurate calculations in the other case
